wemake_python_styleguide/violations/oop.py

```diff
@@ -671,3 +673,32 @@ def get_attribute(self):
 
     error_template = 'Found unpythonic getter or setter'
     code = 615
+
+
+class BuggySuperContextViolation(ASTViolation):
+    """
+    Calling super() in buggy context.
+
+    Reasoning:
+        Call to `super()` without arguments will cause unexpected `TypeError`
+        in a number of specific contexts, e.g. dict/set/list comprehensions
+        and generator expressions.
+
+        Read more: https://bugs.python.org/issue46175
+
+    Solution:
+        Use `super(cls, self)` instead in those cases.
+
+    Example::
+
+        # Correct
+        (super(cls, self).augment(it) for it in items)
+
+        # Wrong
+        (super().augment(it) for it in items)
+
+    .. versionadded:: 0.17.0
```

Comment:
```suggestion
    .. versionadded:: 0.18.0
```

Can you please check other your violations for this?

Reply:
Yeah, I marked them as 0.17.0. Will fix it in a separate PR.